### PR TITLE
Parse transaction role declarations

### PR DESCRIPTION
**Pull request by turbolent**
Work towards #2177

## Description

- Add AST element for transaction role declaration
- Parse transaction role declarations
- Also: Simplify original transaction declaration parsing, remove unnecessary duplicated execute block parsing

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/parser/transaction.go [hunk 1]
```diff
@@ -71,76 +72,65 @@ func parseTransactionDeclaration(p *parser, docString string) (*ast.TransactionD
...
 
-	// Prepare (optional) or execute (optional)
+	// Prepare (optional)
 
 	var prepare *ast.SpecialFunctionDeclaration
-	var execute *ast.SpecialFunctionDeclaration
 
 	p.skipSpaceAndComments()
-	if p.current.Is(lexer.TokenIdentifier) {
+	if p.isToken(p.current, lexer.TokenIdentifier, keywordPrepare) {
```
Comment (SupunS) [inline runtime/parser/transaction.go:80]: Nice, this is much cleaner: the order of blocks is much more intuitive now

File: runtime/parser/declaration_test.go [hunk 2]
```diff
@@ -3731,6 +3731,144 @@ func TestParseTransactionDeclaration(t *testing.T) {
...
+		result, errs := testParseDeclarations(`
+          transaction {
+            role buyer {
+              var x: Int
+
+		      prepare(signer: AuthAccount) {
+	            x = 0
+			  }
+            }
+          }
```
Comment (SupunS) [inline runtime/parser/declaration_test.go:3748]:
```suggestion
          transaction {
            role buyer {
              var x: Int

              prepare(signer: AuthAccount) {
                x = 0
              }
            }
          }
```

File: runtime/parser/declaration_test.go [hunk 3]
```diff
@@ -3731,6 +3731,144 @@ func TestParseTransactionDeclaration(t *testing.T) {
 			result.Declarations(),
 		)
 	})
+
+	t.Run("role", func(t *testing.T) {
```
Comment (SupunS) [inline runtime/parser/declaration_test.go:3735]: maybe also add some tests for role blocks without prepare, empty role blocks, etc.

**Review by SupunS — APPROVED** [review]

File: runtime/parser/declaration_test.go (hunk 3 above)
Comment (dsainati1) [reply runtime/parser/declaration_test.go:3735]: Yea; would like to see tests for the failure cases here. In addition to what Supun mentioned above, would be good to have tests for nested role blocks, role blocks with conditions, and multiple role blocks in a transaction.

Comment (turbolent) [reply runtime/parser/declaration_test.go:3735]: Added more tests in 67105eb570ec74e27c045912b4aa3ea780a4e179.

Nesting and pre/post conditions are not supported

**Review by SupunS — APPROVED** [review]
